src/Agent.Worker/JobExtension.cs

```diff
@@ -238,6 +238,25 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                     }
                     context.Output("Finished checking job knob settings.");
 
+                    // Ensure that we send git telemetry before potential path env changes during the pipeline execution
+                    var isSelfHosted =  StringUtil.ConvertToBoolean(jobContext.Variables.Get(Constants.Variables.Agent.IsSelfHosted));
```

Comment:
Nit: you can use [GetBoolean](https://github.com/microsoft/azure-pipelines-agent/blob/fd1e0a55b060f2cc0640a367a058831b5cbb5230/src/Agent.Worker/Variables.cs#L344) from `Variables`

Reply:
Since the IsSelfHosted constraint is not bool, but string with values "1" and "0" I think we should rework it in the separate PR instead of modifying the generalized GetBoolean method in this PR with unrelated scope. Thanks!